### ISSUE TITLE
Unable to generate kubeconfig after enabling Authorized Cluster Endpoint on RKE2 vSphere Cluster

### ISSUE THREAD

**Issue by mscbpi** [rancher/rancher#38284]
**Rancher Server Setup**
- Rancher version: v2.6.6
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2 v1.23.6
- Proxy/Cert Details: `private-CA`

**Information about the Cluster**
- Kubernetes version: v1.23.7+rke2r2
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):
   Infrastructure Provider: vSphere

**User Information**
- What is the role of the user logged in? Admin
  - If custom, define the set of permissions:

   
**Describe the bug**

https://github.com/rancher/rancher/issues/34500 seems to appear again / not solved for vSphere provider downstream cluster

TL;DR: enabling ACE on a cluster that has not been created with ACE results in a broken `generateKubeconfig`

**To Reproduce**

- vSphere provisioner (provisioned with Rancher)
- Created without ACE
- Enable ACE
- (Try to) download `KUBECONFIG` with the GUI or `generateKubeconfig` API

**Result**

```
{
"baseType": "error",
"code": "ServerError",
"message": "the server could not find the requested resource (post clusterauthtokens.meta.k8s.io)",
"status": 500,
"type": "error"
}
```
**Expected Result**

Generated `KUBECONFIG` with additional contexts.

**Screenshots**

**Additional context**

ACE works since creating a manual `KUBECONFIG` with IP of a master node and `certificate-authority-data` taken from a master `node /etc/rancher/rke2.yaml` works well...

Disabling ACE makes `KUBECONFIG` available again.

Weird enough, my API Endpoint on https://master-node-IP:6443 is still accessible with local `ServiceAccount` bypassing Rancher auth. Have I missed something there ?

Question : what is the policy for closed issues that seems to be unresolved ? I don't know if it is better to comment the closed one or re-open.

**Comment by mscbpi** [comment]
`kube-api-auth` is deployed, maybe explaining the endpoint is available despite having `localClusterAuthEndpoint` set to `false`, but the log reports something wrong:

```E0715 07:37:57.634068 1 reflector.go:139] pkg/mod/github.com/rancher/client-go@v1.22.3-rancher.1/tools/cache/reflector.go:168: Failed to watch *v3.ClusterUserAttribute: failed to list *v3.ClusterUserAttribute: the server could not find the requested resource (get clusteruserattributes.meta.k8s.io)```

**Comment by volenbovsky** [comment]
It seems that I had similar issue and the workaround was to restart Rancher PODs
(my provisioner is not a vSphere, but rather 'bare metal')

**Comment by mscbpi** [comment]
Which one, `cattle-agent` ?

**Comment by volenbovsky** [comment]
@mscbpi , cattle-agent is something in Downstream Cluster.
I mean _Rancher_ the thing that processes the request of providing kubeconfig file (implemented eg. as PODs, typically in other Kubernetes cluster)

**Comment by mscbpi** [comment]
@volenbovsky 
1. create downstream cluster with ACE disabled using vSphere provider
2. Enable ACE on the cluster (GUI or YAML, Cluster or ManagementCluster object)
==> `KUBECONFIG` fails 
3. Disable ACE
==> `KUBECONFIG` works (cluster URL through rancher endpoint only)

4. Enable ACE again
5. Restart `rancher` pods on management cluster
==> `KUBECONFIG` generated with rancher + master nodes cluster endpoints.

So you are right, thanks !

however this is still a glitch/bug, don't you think ?

**Comment by mscbpi** [comment]
The issue is still relevant

**Comment by ron1** [comment]
The issue is still relevant.
